ios/instruments/helper.go

```diff
@@ -29,7 +30,11 @@ func connectInstruments(device ios.DeviceEntry) (*dtx.Connection, error) {
 		log.Debugf("Failed connecting to %s, trying %s", serviceName, serviceNameiOS14)
 		dtxConn, err = dtx.NewUsbmuxdConnection(device, serviceNameiOS14)
 		if err != nil {
-			return nil, err
+			log.Debugf("Failed connecting to %s, trying %s", serviceNameiOS14, serviceNameiOS17)
+			dtxConn, err = dtx.NewTunnelConnection(device, serviceNameiOS17)
+			if err != nil {
+				return nil, err
+			}
```

Comment:
TBH looks not great with iOS 17 as a fallback, probably a proper check for iOS version would avoid potentially unlimited nesting :)

Reply:
actuall this part is solved already in https://github.com/danielpaulus/go-ios/pull/396 but unfortunately it's not merged yet, and I can't rebase someone else's PR. Maybe pinging @michal-przytarski works and he can rebase, after that I can merge it.